### ISSUE TITLE
False positive when using interface + trait

### ISSUE THREAD

**Issue by enumag** [phpstan/phpstan-strict-rules#48]
Testing code:

```php
interface FooInterface
{
	/**
	 * @param mixed $value
	 *
	 * @return mixed
	 */
	public function test($value);
}

trait FooTrait
{
	/**
	 * @param mixed $value
	 *
	 * @return mixed
	 */
	public function test($value)
	{
	}
}

class Bar implements FooInterface
{
	use FooTrait;
}
```

Actual output:

```
 ------ ------------------------------------------------------------------------- 
  Line   Bar.php (in context of class App\Bar)                                 
 ------ ------------------------------------------------------------------------- 
  204    Method App\Bar::test() has no return typehint specified.                 
  204    Method App\Bar::test() has parameter $value with no typehint specified.  
 ------ ------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 2 errors

```

Expected output: no error, both the parameter and the return type are correctly annotated.

**Comment by pkollitsch** [comment]
What is in line 204, the class Bar or the trait FooTrait?

**Comment by enumag** [comment]
FooTrait

**Comment by enumag** [comment]
This seems to be fixed now.
